test/test_client.py

```diff
@@ -535,6 +535,18 @@ def test_client_options(self):
         self.assertIsInstance(c.options.retry_writes, bool)
         self.assertIsInstance(c.options.retry_reads, bool)
 
+    def test_validate_suggestion(self):
+        """Validate kwargs in constructor."""
+        self.assertRaises(ConfigurationError, MongoClient, auth='standard')
+
+        try:
+            MongoClient(auth='standard')
+        except ConfigurationError as exc:
+            expected = "Unknown option: auth. Did you mean: authsource, authmechanism, authoidcallowedhosts?"
```

Comment:
Would it be possible to change this to:
```
Did you mean: authSource, authMechanism, ...?"
```

Reply:
Upon investigation, updating all the cases of the piles and piles of keywords in these dicts, finding the matching ones, etc etc isn't worth it. It's a lot of work, and it's brittle. 
I have a much simpler proposal. I've added a note to my error message about camelCase. What do you think?
ConfigurationError('Unknown option: auth. Did you mean one of (authsource, authmechanism, authoidcallowedhosts) **or maybe a camelCase version of one?'**)

Reply:
Okay that's fair. Although I suspect there is a non-brittle way to do it but it would require more work than needed for this PR. Could you open this idea as a new PYTHON ticket so we can reevaluate later on?